tests/Microsoft.Identity.Test.Unit/PublicApiTests/ClientCredentialWithCertTest.cs

```diff
@@ -100,7 +100,7 @@ public async Task JsonWebTokenWithX509PublicCertSendCertificateTestAsync()
                     .WithAuthority(new System.Uri(ClientApplicationBase.DefaultAuthority), true)
                     .WithRedirectUri(TestConstants.RedirectUri)
                     .WithHttpManager(harness.HttpManager)
-                    .WithCertificate(certificate)
+                    .WithCertificate(certificate, true)
```

Comment:
We need unit tests that cover all combinations: 

| app |  request | 
| -----|  --- |
| true | false |
| false | true |
| true | not set |
| false | not set |